h/static/scripts/controllers/confirm-submit-controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var Controller = require('../base/controller');
+
+/* Turn a normal submit element into one that shows a confirm dialog.
+ *
+ * The element's form will only be submitted if the user answers the
+ * confirmation dialog positively.
+ *
+ */
+class ConfirmSubmitController extends Controller {
```

Comment:
As with another controller I wrote recently (copy clipboard button), this isn't a "React-style" controllers. It's not using update, state, etc. This is partially because I haven't gotten the hang of / internalised that pattern yet. But it's also because these controllers are so trivial, needing only a constructor function that adds an event listener, that it doesn't seem necessary.